File: backend/Domain/Entities/Warehouse.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities
{
    public class Warehouse : BaseEntity
    {
        [ForeignKey(nameof(CommercialProperty))]
        public Guid CommercialPropertyId { get; set; }
        public double CeilingHeight { get; set; }
        public bool LoadingDockAvailable { get; set; }
        public bool OfficeSpaceAvailable { get; set; }
        [MinLength(1)]
        public List<WarehouseSuitableGoods> SuitableGoods { get; set; } = new();

        public CommercialProperty CommercialProperty { get; set; } = null!;
    }
}

File: mobile/lib/src/features/dashboard/domain/entities/warehouse.dart
import 'package:equatable/equatable.dart';
import 'package:mobile/src/features/dashboard/domain/domain.dart';

class Warehouse extends Equatable {
  final double ceilingHeight;
  final bool loadingDockAvailable;
  final bool officeSpaceAvailable;
  final List<WarehouseSuitableGoods>? suitableGoods;

  const Warehouse({
    required this.ceilingHeight,
    required this.loadingDockAvailable,
    required this.officeSpaceAvailable,
    this.suitableGoods,
  });
  @override
  List<Object?> get props => [
        ceilingHeight,
        loadingDockAvailable,
        officeSpaceAvailable,
        suitableGoods
      ];
}
